File: src/main/java/com/example/onemoretimejava/arrays/FindTheHighestAltitude.java
package com.example.onemoretimejava.arrays;

public class FindTheHighestAltitude {
    public static void main(String[] args) {
       /* Input: gain = [-5,1,5,0,-7]
        Output: 1*/
        int[] arr = {-5, 1, 5, 0, -7};
        largestAltitude(arr);
    }

    public static int largestAltitude(int[] arr) {
        int ans = 0;
        int n = arr.length;
        int res = 0;
        for (int i = 0; i < n; i++) {
            ans = ans + arr[i];
            res = Math.max(res, ans);
        }
        System.out.println(res);
        return res;
    }
}

File: src/main/java/com/example/onemoretimekotlin/arrays/FindTheHighestAltitude.kt
package com.example.onemoretimekotlin.arrays

fun main(args: Array<String>) {
    val arr = intArrayOf(-5, 1, 5, 0, -7)
    largestAltitude(arr)

}

fun largestAltitude(gain: IntArray): Int {
    var ans = 0;
    var res = 0;
    for (i in 0 until gain.size) {
        ans += gain[i];
        res = Math.max(res, ans)
    }
    println(res)
    return res
}